File: OPENAPI-REST-API/openapi-client/scala-gatling/src/gatling/scala/org/openapitools/client/model/TicketDetails.scala

package org.openapitools.client.model


case class TicketDetails (
    _ticketid: Option[Integer],
    _ticketmaskid: Option[String],
    _department: Option[String],
    _status: Option[String],
    _priority: Option[String],
    _subject: Option[String],
    _created_on: Option[String],
    _updated_on: Option[String]
)
object TicketDetails {
    def toStringBody(var_ticketid: Object, var_ticketmaskid: Object, var_department: Object, var_status: Object, var_priority: Object, var_subject: Object, var_created_on: Object, var_updated_on: Object) =
        s"""
        | {
        | "ticketid":$var_ticketid,"ticketmaskid":$var_ticketmaskid,"department":$var_department,"status":$var_status,"priority":$var_priority,"subject":$var_subject,"created_on":$var_created_on,"updated_on":$var_updated_on
        | }
        """.stripMargin
}

File: OPENAPI-REST-API/openapi-client/zapier/models/TicketDetails.js
const utils = require('../utils/utils');

module.exports = {
    fields: (prefix = '', isInput = true, isArrayChild = false) => {
        const {keyPrefix, labelPrefix} = utils.buildKeyAndLabel(prefix, isInput, isArrayChild)
        return [
            {
                key: `${keyPrefix}ticketid`,
                label: `[${labelPrefix}ticketid]`,
                type: 'integer',
            },
            {
                key: `${keyPrefix}ticketmaskid`,
                label: `[${labelPrefix}ticketmaskid]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}department`,
                label: `[${labelPrefix}department]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}status`,
                label: `[${labelPrefix}status]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}priority`,
                label: `[${labelPrefix}priority]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}subject`,
                label: `[${labelPrefix}subject]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}created_on`,
                label: `[${labelPrefix}created_on]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}updated_on`,
                label: `[${labelPrefix}updated_on]`,
                type: 'string',
            },
        ]
    },
    mapping: (bundle, prefix = '') => {
        const {keyPrefix} = utils.buildKeyAndLabel(prefix)
        return {
            'ticketid': bundle.inputData?.[`${keyPrefix}ticketid`],
            'ticketmaskid': bundle.inputData?.[`${keyPrefix}ticketmaskid`],
            'department': bundle.inputData?.[`${keyPrefix}department`],
            'status': bundle.inputData?.[`${keyPrefix}status`],
            'priority': bundle.inputData?.[`${keyPrefix}priority`],
            'subject': bundle.inputData?.[`${keyPrefix}subject`],
            'created_on': bundle.inputData?.[`${keyPrefix}created_on`],
            'updated_on': bundle.inputData?.[`${keyPrefix}updated_on`],
        }
    },
}
